glib-macros/src/object_subclass_attribute.rs

```diff
@@ -0,0 +1,88 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use proc_macro2::TokenStream;
+use quote::quote;
+
+pub fn impl_object_subclass(input: &syn::ItemImpl) -> TokenStream {
+    let mut has_new = false;
```

Comment:
You probably want to check that this is only added to an `impl .*ObjectSubclass for T` and not on anything else

Reply:
Technically someone could `use glib::ObjectSubclass as FooBar`, and we'd have no way to know that. But yeah, it's probably reasonable to restrict it to `ObjectSubclass` or paths ending in that.

Reply:
Well, at least check that it's some trait impl block :)